clang-tools-extra/clangd/ParsedAST.cpp

```diff
@@ -451,6 +451,7 @@ ParsedAST::build(llvm::StringRef Filename, const ParseInputs &Inputs,
   DiagnosticConsumer *DiagConsumer = &ASTDiags;
   IgnoreDiagnostics DropDiags;
   if (Preamble) {
+    CI->TargetOpts = Preamble->TargetOpts;
     Patch = PreamblePatch::createFullPatch(Filename, Inputs, *Preamble);
```

Comment:
Idea: should we make this part of `PreamblePatch`?
It seems to fit the concept (take inputs from preamble, patch up compiler invocation based on it). Although I'm not sure if I'm misreading its intention.

Reply:
well, idea also crossed my mind while putting together the patch but wasn't so sure if that would be the best place. since preamble-patch'ing was mostly for adjusting the compilation to "patch" changes in the preamble contents.

but considering the fact that this actually needs to be done on all the places that re-uses a preamble (CodeCompletion, SignatureHelp), I guess PreamblePatch is a nice common ground (I was leaning towards `prepareCompilerInstance` but unfortunately that one can't depend on Preamble.h :/)